src/Compiler/pars.fsy

```diff
@@ -1271,6 +1271,14 @@ moduleDefnsOrExpr:
 
 /* A sequence of definitions in a namespace or module */
 moduleDefns:
+  | moduleDefnOrDirective moduleDefnOrDirective
+      { match $1, $2 with
+        | [SynModuleDecl.Types _], [SynModuleDecl.NestedModule(_, _, _, _, _, { ModuleKeyword = Some m })] ->
```

Comment:
We don't need to compare the start column of each one's range?

Reply:
Yes we do :). The column comparison will make the error detection much more accurate and reduces false positives.